scico/solver.py

```diff
@@ -710,14 +715,27 @@ def __init__(
         """
         if isinstance(A, MatrixOperator):
             A = A.to_array()
+        elif not isinstance(A, Array):
```

Comment:
I believe that if we implemented `__array__` in `MatrixOperator`, then `jnp.array(A)` would work for either MatrixOperators or JAX arrays. The input dtype could be `array_like`. Similar for `D`. Might be cleaner/less coupled to `MatrixOperator` if written that way.

Reply:
This seems like a good suggestion. Having not worked with `__array__` myself, perhaps you could help with this?

Reply:
I have addressed this in a commit. Please check, especially the new text in the arguments section of the docs.

Reply:
Looks good.